core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java

```diff
@@ -1,56 +1,41 @@
 package com.scalar.db.api;
 
-import com.scalar.db.exception.transaction.CommitConflictException;
 import com.scalar.db.exception.transaction.CrudConflictException;
 import com.scalar.db.exception.transaction.CrudException;
-import com.scalar.db.exception.transaction.PreparationConflictException;
 import com.scalar.db.exception.transaction.UnsatisfiedConditionException;
 import java.util.List;
 import java.util.Optional;
 
-/**
- * An interface for transactional CRUD operations. Note that the LINEARIZABLE consistency level is
- * always used in transactional CRUD operations, so {@link Consistency} specified for CRUD
- * operations is ignored.
- */
-public interface TransactionCrudOperable {
+/** An interface for transactional CRUD operations for transactions. */
+public interface TransactionCrudOperable extends CrudOperable<CrudException> {
```

Comment:
This might be just a naming issue. IIUC, `CrudException` inherits from `TransactionException`, and `TransactionCrudOperable` inherits from `CrudOperable`. Their inheritance directions seem reversed from the perspective of class names. It's a bit confusing?

Reply:
Good point. Thanks.

Actually, this naming strongly depends on backward compatibility. `TransactionCrudOperable` is an existing name, so we can't rename it to preserve backward compatibility.

The requirements here are as follows:

- Add CRUD operation methods to the transaction manager interfaces (`DistributedTransactionManager` and `TwoPhaseCommitTransactionManager`).
- Access the CRUD operation methods through the same abstraction for the transaction manager interfaces and the transaction interfaces (`DistributedTransaction` and `TwoPhaseCommitTransaction`).
- The exceptions thrown by the CRUD operation methods differ slightly between the transaction manager interfaces and the transaction interfaces.

Considering these requirements, we finally decided on the current naming. But do you have any better ideas for the naming?

Reply:
Thanks for sharing the background.

What we can change is only the name of `CrudOperable` newly added in this PR. It seems to be only inherited by `TransactionManagerCrudOperable` and `TransactionCrudOperable`. How about renaming it to `TransactionCrudOperableBase` or something so that the inheritance relationships will be as follows?
- `TransactionCrudOperableBase`
  - `TransactionCrudOperable`
  - `TransactionManagerCrudOperable`
- `TransactionException`
  - `CrudException`

Current
- `CrudOperable`
  - `TransactionCrudOperable`
  - `TransactionManagerCrudOperable`
- `TransactionException`
  - `CrudException`

Reply:
Actually, `TransactionManagerCrudOperable` is also newly added in this PR, so we can rename it as well.

The naming `XXXBase` sounds more appropriate for abstract classes to me, so I'm not sure it's suitable for interfaces. What do you think?

Reply:
I see. I think `TransactionManagerCrudOperable` is alined with `TransactionCrudOperable`, so it can be left as is. Regarding `XxxxBase`, indeed it's less common. I'm also not sure if we should avoid it. Another option is `TransactionalCrudOperable` based on the comment `An interface for transactional CRUD operations`? It might be a bit confusing, though.

Reply:
Yeah, `TransactionalCrudOperable` is a possible idea. But, as you mentioned, it might still be confusing between `TransactionalCrudOperable` and `TransactionCrudOperable`.

If we consider `CrudOperable` vs. `TransactionalCrudOperable`, I prefer `CrudOperable`. But that's just my preference. If you have any other better ideas, strong opinions, or strong reasons, please let me know. Thanks.